src/com/projectkorra/projectkorra/airbending/combo/AirSweep.java

```diff
@@ -131,22 +132,22 @@ public void progress() {
 
 		if (this.origin == null) {
 			this.direction = this.player.getEyeLocation().getDirection().normalize();
-			this.origin = this.player.getLocation().add(this.direction.clone().multiply(10));
+			this.origin = GeneralMethods.getMainHandLocation(player).add(this.direction.clone().multiply(10));
 		}
 		if (this.progressCounter < 8) {
 			return;
 		}
 		if (this.destination == null) {
-			this.destination = this.player.getLocation().add(this.player.getEyeLocation().getDirection().normalize().multiply(10));
+			this.destination = GeneralMethods.getMainHandLocation(player).add(GeneralMethods.getMainHandLocation(player).getDirection().normalize().multiply(10));
 			final Vector origToDest = GeneralMethods.getDirection(this.origin, this.destination);
 			for (double i = 0; i < 30; i++) {
 				final Location endLoc = this.origin.clone().add(origToDest.clone().multiply(i / 30));
-				if (GeneralMethods.locationEqualsIgnoreDirection(this.player.getLocation(), endLoc)) {
+				if (GeneralMethods.locationEqualsIgnoreDirection(GeneralMethods.getMainHandLocation(player), endLoc)) {
```

Comment:
You should probably make a variable at the beginning of the loop for getMainHandLocation so it isn't run 3 times